de.dlr.sc.overtarget.language.ui/src/de/dlr/sc/overtarget/language/ui/handler/GenerationHandler.java

```diff
@@ -80,10 +82,13 @@ public CancelIndicator getCancelIndicator() {
 			final EclipseResourceFileSystemAccess2 fsa = fileSystemAccessProvider.get();
 			fsa.setMonitor(new NullProgressMonitor());
 			
-			Set<OutputConfiguration> overtargetConfigurations = new OvertargetOutputConfigurationProvider().getOutputConfigurations();
-			fsa.getOutputConfigurations().put(OvertargetOutputConfigurationProvider.GENERATOR_OUTPUT_ID_OVERTARGET, overtargetConfigurations.iterator().next());
+			final EclipseOutputConfigurationProvider configProvider = eclipseOutputConfigProvider.get();
+			String configValue = configProvider.getPreferenceStoreAccess().getContextPreferenceStore(project).getString("outlet.de.dlr.sc.overtarget.output.directory");
+			configProvider.getOutputConfigurations().iterator().next().setOutputDirectory(configValue);
+			fsa.setOutputPath(configValue);
+			fsa.setOutputPath("de.dlr.sc.overtarget.output", configValue);
```

Comment:
OK, actually... Three times? :D Do we need that or did you just try out which one works and then didn't clean up?